packages/asset-buyer/test/forwarder_swap_quote_consumer_test.ts

```diff
@@ -0,0 +1,142 @@
+        ];
+    });
+    after(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+    beforeEach(async () => {
+        await blockchainLifecycle.startAsync();
+        // This constructor has incorrect types
+    });
+    afterEach(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+    describe('getSmartContractParamsOrThrow', () => {
+
+        describe('validation', () => {
+            it('should throw if swap quote provided is not a valid forwarder SwapQuote (taker asset is WETH)', async () => {
+                const invalidSignedOrders = getSignedOrdersWithNoFees(
+                    makerAssetData,
+                    takerAssetData,
+                    makerAddress,
+                    takerAddress,
+                    FILLABLE_AMOUNTS,
+                );
+                const invalidSwapQuote = getFullyFillableSwapQuoteWithNoFees(makerAssetData, takerAssetData, invalidSignedOrders);
+                const swapQuoteConsumer = new ForwarderSwapQuoteConsumer(provider, {});
+                // TODO(dave4506) finish up testing/coverage
+                // expect(
+                //     swapQuoteConsumer.getSmartContractParamsOrThrow(invalidSwapQuote, {}),
+                // ).to.throws();
+            });
+        });
+
+        describe('valid swap quote', async () => {
+            it('provide correct smart contract params with default options', async () => {
+                const signedOrders = getSignedOrdersWithNoFees(
+                    makerAssetData,
+                    wethAssetData,
+                    makerAddress,
+                    takerAddress,
+                    FILLABLE_AMOUNTS,
+                );
+                const swapQuote = getFullyFillableSwapQuoteWithNoFees(makerAssetData, takerAssetData, signedOrders);
+                const swapQuoteConsumer = new ForwarderSwapQuoteConsumer(provider, { networkId });
+                const smartContractParamsInfo = await swapQuoteConsumer.getSmartContractParamsOrThrowAsync(swapQuote, {});
+                // console.log(smartContractParamsInfo);
+                // TODO(dave4506): Add elaborate testing
+            });
+        });
+    });
+
+    describe('getCalldataOrThrow', () => {
+
+        describe('validation', () => {
+            it('should throw if swap quote provided is not a valid forwarder SwapQuote (taker asset is WETH)', async () => {
+                const invalidSignedOrders = getSignedOrdersWithNoFees(
+                    makerAssetData,
+                    takerAssetData,
+                    makerAddress,
+                    takerAddress,
+                    FILLABLE_AMOUNTS,
+                );
+                const invalidSwapQuote = getFullyFillableSwapQuoteWithNoFees(makerAssetData, takerAssetData, invalidSignedOrders);
+                const swapQuoteConsumer = new ForwarderSwapQuoteConsumer(provider, {});
+                // TODO(dave4506) finish up testing/coverage
```

Comment:
Is this intentional?

Reply:
if it helps you need to pass in a function that throws, not an expression that throws.

Reply:
This is corrected in another PR: https://github.com/0xProject/0x-monorepo/pull/1896